framework/h5py_interface_creator.py

```diff
@@ -302,10 +308,12 @@ def addGroupInit(self,groupName,attributes=None):
     grp.attrs[b'groupName'] = groupNameInit
     self.allGroupPaths.append(utils.toBytes("/" + groupNameInit))
     self.allGroupEnds.append(False)
-    self.h5FileW["allGroupPaths"].resize((len(self.allGroupPaths),))
-    self.h5FileW["allGroupEnds"].resize( (len(self.allGroupEnds),) )
-    self.h5FileW["allGroupPaths"][...] = self.allGroupPaths
-    self.h5FileW["allGroupEnds"][...] = self.allGroupEnds
+    if len(self.allGroupPaths) > len(self.h5FileW["allGroupPaths"]):
+      self.h5FileW["allGroupPaths"].resize((len(self.allGroupPaths)*2,))
+      self.h5FileW["allGroupEnds"].resize( (len(self.allGroupPaths)*2,) )
+    self.h5FileW["allGroupPaths"][len( self.allGroupPaths) - 1] = self.allGroupPaths[-1]
+    self.h5FileW["allGroupEnds"][len(self.allGroupPaths) - 1] = self.allGroupEnds[-1]
+    self.h5FileW.attrs["nGroups"] = len(self.allGroupPaths)
```

Comment:
A function may be useful since these changes are repeated

Reply:
added